tests/SpecTests/ClientSideEncryption/Prose25_LookupTest.php

```diff
@@ -0,0 +1,379 @@
+            ],
+        ];
+
+        $this->assertPipelineReturnsSingleDocument(self::COLL_NO_SCHEMA, $pipeline, $expected);
+    }
+
+    public function testCase5_CsfleJoinsCsfle2(): void
+    {
+        $pipeline = [
+            ['$match' => ['csfle' => 'csfle']],
+            [
+                '$lookup' => [
+                    'from' => 'csfle2',
+                    'as' => 'matched',
+                    'pipeline' => [
+                        [
+                            '$match' => ['csfle2' => 'csfle2'],
+                        ],
+                        [
+                            '$project' => ['_id' => 0],
+                        ],
+                    ],
+                ],
+            ],
+            ['$project' => ['_id' => 0]],
+        ];
+        $expected = ['csfle' => 'csfle', 'matched' => [['csfle2' => 'csfle2']]];
+
+        $this->assertPipelineReturnsSingleDocument(self::COLL_CSFLE, $pipeline, $expected);
+    }
+
+    public function testCase6_QeJoinsQe2(): void
+    {
+        $pipeline = [
+            ['$match' => ['qe' => 'qe']],
+            [
+                '$lookup' => [
+                    'from' => 'qe2',
+                    'as' => 'matched',
+                    'pipeline' => [
+                        [
+                            '$match' => ['qe2' => 'qe2'],
+                        ],
+                        [
+                            '$project' => [
+                                '_id' => 0,
+                                '__safeContent__' => 0,
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+            ['$project' => ['_id' => 0, '__safeContent__' => 0]],
+        ];
+        $expected = ['qe' => 'qe', 'matched' => [['qe2' => 'qe2']]];
+
+        $this->assertPipelineReturnsSingleDocument(self::COLL_QE, $pipeline, $expected);
+    }
+
+    public function testCase7_NoSchemaJoinsNoSchema2(): void
+    {
+        $pipeline = [
+            ['$match' => ['no_schema' => 'no_schema']],
+            [
+                '$lookup' => [
+                    'from' => 'no_schema2',
+                    'as' => 'matched',
+                    'pipeline' => [
+                        ['$match' => ['no_schema2' => 'no_schema2']],
+                        ['$project' => ['_id' => 0]],
+                    ],
+                ],
+            ],
+            ['$project' => ['_id' => 0]],
+        ];
+        $expected = ['no_schema' => 'no_schema', 'matched' => [['no_schema2' => 'no_schema2']]];
+
+        $this->assertPipelineReturnsSingleDocument(self::COLL_NO_SCHEMA, $pipeline, $expected);
+    }
+
+    public function testCase8_CsfleJoinsQeFails(): void
+    {
+        $this->skipIfServerVersion('<', '8.1.0', 'Lookup test case requires server version 8.1.0 or later');
+        $this->skipIfClientSideEncryptionIsNotSupported();
+
+        $this->expectExceptionMessageMatches('#not supported#');
```

Comment:
I think `expectExceptionMessage` already matches substrings. `expectExceptionMessageMatches` is only if you have a regex.
```suggestion
        $this->expectExceptionMessage('not supported');
```